### ISSUE TITLE
Global Mode

### ISSUE THREAD

**Issue by GaoMax** [sheng/ladder-ios#20]
Is it possible to add a global mode, so that all the traffic gets proxied ?

**Comment by sheng** [comment]
This project uses [PAC](https://en.wikipedia.org/wiki/PAC) to decide when to transfer. So, it is not necessary to add a "Global Mode".

In fact, you can host a custom PAC file with the following content:

```javascript
function FindProxyForURL(url, host) {
	return "SOCKS5 127.0.0.1:1081; SOCKS 127.0.0.1:1081; DIRECT;";
}
```

This way all your traffic will be transferred (just like the Global Mode).

**Comment by Taherismail17** [comment]
Doesn’t work with Facebook App but works with Facebook website - Potatso Client lets it work with both. There is an app issue. 

Firewall - Fortigate

**Comment by sheng** [comment]
@Taherismail17 

Yep, there is an app issue that will cause your problem. This project currently only supports `HTTP(S)` proxies. Pure `TCP` and `UDP` proxies will be implemented in the version `2.0.0`, and I'm working on it.

**Comment by Taherismail17** [comment]
I have followed this tutorial to set up my backend, 
https://microfish.blog/2017/05/22/set-up-shadowsocks-vpn-in-an-ubuntu-vm-on-azure/

Any idea how I can create a fix or when do you think you would be able to to implement this fix?

**Comment by sheng** [comment]
@Taherismail17 

I'm not sure when I can fix it. There is no detailed plan. But I already have the basic idea, so I will do my best to fix it ASAP.

**Comment by Taherismail17** [comment]
Awesome, thanks. My VPN doesn’t work behind Fortigate and this should solve it. 

Thanks

**Comment by Taherismail17** [comment]
Does the latest commit add the TCP and UDP proxies with the help of Lwip?

**Comment by sheng** [comment]
@Taherismail17 

Yes, I intend to use [lwIP](https://en.wikipedia.org/wiki/LwIP) to convert `TUN` to `SOCKS` to proxy requests for various protocols.

**Comment by Taherismail17** [comment]
@sheng Is it currently functioning?

**Comment by sheng** [comment]
@Taherismail17 

Not yet, I'm still working on it.

**Comment by sheng** [comment]
@Taherismail17 

I did not test it. According to [NEKit's README](https://github.com/zhuhaow/NEKit/blob/bdf8f5fd444a1829173f63a416b88240324c9db7/README.md), the `TUNInterface` is broken.

**Comment by Taherismail17** [comment]
@Sheng I tried it and the VPN disconnected when on I opened up the Instagram and Facebook apps. It did work with the website.
